dbms/src/Storages/StorageFile.cpp

```diff
@@ -254,6 +276,7 @@ class StorageFileBlockInputStream : public IBlockInputStream
 
 private:
     std::shared_ptr<StorageFile> storage;
+    std::optional<std::string> file_path;
```

Comment:
Maybe we can remove `std::optional` and use empty string for default value.

Reply:
Is explicit better than implicit?